src/main.rs

```diff
@@ -29,21 +26,19 @@ use serde_json::value::Value as ReportBody;
 /// and passes it to any matching request handlers before returning a `Response`
 /// back downstream.
 #[fastly::main]
-fn main(req: Request<Body>) -> Result<Response<Body>, Error> {
+fn main(req: Request) -> Result<Response, Error> {
     // Pattern match on the request method and path.
-    match (req.method(), req.uri().path()) {
-        // If a CORS preflight OPTIONS request return a 204 no content.
-        (&Method::OPTIONS, "/report") => generate_no_content_response(),
-        // If a POST request pass to the `handler_reports` request handler.
+    match (req.get_method(), req.get_path()) {
+        // For a CORS preflight OPTIONS request, return an empty 204 response.
+        (&Method::OPTIONS, "/report") => Ok(generate_empty_204_response()),
+        // Pass a POST request to the `handler_reports` request handler.
         (&Method::POST, "/report") => {
-            let _ = handle_reports(req);
-            // Return an empty 204 No Content response to the downstream client.
-            generate_no_content_response()
+            let _ = handle_reports(req)?;
+            // Return an empty 204 response to the downstream client.
+            Ok(generate_empty_204_response())
         }
-        // For all other requests return a 404 not found.
-        _ => Ok(Response::builder()
-            .status(StatusCode::NOT_FOUND)
-            .body(Body::from("Not found"))?),
+        // For all other requests return a 404.
+        _ => Ok(Response::from_body("Not found").with_status(StatusCode::NOT_FOUND)),
```

Comment:
Apologies, I thought I had written this into my last review.

Plain text responses should generally have trailing newlines.  And we also have a handy helper function for such cases which sets an appropriate `Content-Type: text/plain; charset=utf-8`. (this is contrary to the docs which incorrectly say it sets "text/html", and that is going to be fixed).

```suggestion
        _ => Ok(Response::from_status(StatusCode::NOT_FOUND).with_body_str("Not found\n")),
```

Reply:
The newline at the end seems counter-intuitive. I wonder if this should be left to the function implementation.